backend/src/components/lookup.js

```diff
@@ -55,31 +55,32 @@ const fundingModelType = [
   },
 ];
 
-function parseProgramYear(value) {
+function parseProgramYear(programYearList) {
   const programYears = {
-    current: undefined,
-    future: undefined,
-    previous: undefined,
     renewal: undefined,
     newApp: undefined,
     list: [],
   };
-  value.forEach((item) => {
-    const p = new MappableObjectForFront(item, ProgramYearMappings).data;
-    const currentStatus = p.status;
-    p.status = getLabelFromValue(p.status, PROGRAM_YEAR_STATUS_CODES);
+
+  programYearList.forEach((item) => {
+    const year = new MappableObjectForFront(item, ProgramYearMappings).data;
+    const currentStatus = year.status;
+
+    year.status = getLabelFromValue(year.status, PROGRAM_YEAR_STATUS_CODES);
     if (currentStatus == PROGRAM_YEAR_STATUS_CODES.CURRENT) {
-      programYears.current = p;
+      programYears.current = year;
     } else if (currentStatus == PROGRAM_YEAR_STATUS_CODES.FUTURE) {
-      programYears.future = p;
+      programYears.renewal = year;
```

Comment:
Does this logic not depend on the order of the programYearList? Or does it work because there is only one CURRENT and one FUTURE record?

Reply:
there is only ever one CURRENT and one FUTURE record